cvmfs/history.h

```diff
@@ -165,6 +165,12 @@ class History {
    */
   virtual bool GetHashes(std::vector<shash::Any> *hashes) const = 0;
 
+  // database file management controls
+  virtual void TakeDatabaseFileOwnership() {};
```

Comment:
Wouldn't it be better to make them abstract?  Otherwise one might forget to implement them in a derived class.